### PR TITLE
[TEVA-866/861] current location not working for some users

### PR DESCRIPTION
**Pull request by alexbowen**
https://dfedigital.atlassian.net/browse/TEVA-866
https://dfedigital.atlassian.net/browse/TEVA-861

- refactor the functionality to use native JS instead of jQuery
- add error handling for null response success of postcodes.io request
- some linting autofix spacing has crept in to this PR

## Screenshots of UI changes:

Screenshot 2020-06-24 at 22 27 59

the error message gets removed when the location element get focus again

### CODE REVIEW

**Comment by alexbowen** [comment]
the rspec test failing here i am going to delete and replace with a JS test before this can be merged

File: app/frontend/src/lib/currentLocation.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+};
+
+export const onFailiure = () => {
+  currentLocation.showErrorMessage(document.getElementById('current-location'));
+  disableRadiusSelect();
+  currentLocation.stopLoading(document.querySelector('.js-location-finder'), document.querySelector('.js-location-finder__input'));
+};
+
+export const postcodeFromPosition = (position) => getPostcodeFromCoordinates(position.coords.latitude, position.coords.longitude).then((response) => {
+  if (response.result) {
```
Comment (cpjmcquillan) [inline app/frontend/src/lib/currentLocation.js:60]: Presuming `response.result` is always an array with at least one element if it's there, is it worth making sure?

Comment (alexbowen) [reply app/frontend/src/lib/currentLocation.js:60]: ill get this into the aforementioned test

File: app/frontend/src/lib/currentLocation.test.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    stopLoadingMock = jest.spyOn(currentLocation, 'stopLoading');
+
+    enableRadiusMock = jest.spyOn(radius, 'enableRadiusSelect');
+    disableRadiusMock = jest.spyOn(radius, 'disableRadiusSelect');
+
+    document.body.innerHTML = '<div class="js-location-finder"><input type="text" id="location" class="js-location-finder__input" /></div>';
+  });
+
+  describe('onFaliure', () => {
+    test('returns true if location is in predefined list', () => {
```
Comment (cpjmcquillan) [inline app/frontend/src/lib/currentLocation.test.js:26]: Are these the correct test names?

Comment (alexbowen) [reply app/frontend/src/lib/currentLocation.test.js:26]: ah well spotted, theres probable describe block copy and pasted all over this file :D

**Pull request merged**